subprojects/core/src/main/java/org/gradle/internal/classpath/transforms/JarTransformFactoryForLegacy.java

```diff
@@ -0,0 +1,114 @@
+import java.io.IOException;
+
+@ServiceScope(Scopes.UserHome.class)
+public class JarTransformFactoryForLegacy implements JarTransformFactory {
+
+    private final ClasspathBuilder classpathBuilder;
+    private final ClasspathWalker classpathWalker;
+
+    public JarTransformFactoryForLegacy(ClasspathBuilder classpathBuilder, ClasspathWalker classpathWalker) {
+        this.classpathBuilder = classpathBuilder;
+        this.classpathWalker = classpathWalker;
+    }
+
+    @Override
+    public void applyConfigurationTo(Hasher hasher) {
+        // Do nothing, this is compatible with the old instrumentation
+    }
+
+    @Override
+    public JarTransform createTransformer(File source, ClassTransform classTransform, InstrumentingTypeRegistry typeRegistry) {
+        Boolean isMultiReleaseJar = null;
+
+        if (source.isFile()) {
+            // Walk a file to figure out if it is signed and if it is a multi-release JAR.
+            try (ZipInput entries = FileZipInput.create(source)) {
+                for (ZipEntry entry : entries) {
+                    String entryName = entry.getName();
+                    if (isJarSignatureFile(entryName)) {
+                        // TODO(mlopatkin) Manifest of the signed JAR contains signature information and must be the first entry in the JAR.
+                        //  Looking into the manifest here should be more effective.
+                        // This policy doesn't transform signed JARs so no further checks are necessary.
+                        return new SkipJarTransform(source);
+                    }
+                    if (isMultiReleaseJar == null && JarUtil.isManifestName(entryName)) {
+                        isMultiReleaseJar = JarUtil.isMultiReleaseJarManifest(JarUtil.readManifest(entry.getContent()));
+                    }
+                }
+            } catch (FileException e) {
+                // Ignore malformed archive, let the transformation handle it.
+            } catch (IOException e) {
+                throw new UncheckedIOException(e);
+            }
+        }
+        if (isMultiReleaseJar != null && isMultiReleaseJar) {
+            return new MultiReleaseJarTransformForLegacy(source, classpathBuilder, classpathWalker, typeRegistry, classTransform);
+        }
+        return new BaseJarTransform(source, classpathBuilder, classpathWalker, typeRegistry, classTransform);
+    }
+
+    private boolean isJarSignatureFile(String entryName) {
+        return entryName.startsWith("META-INF/") && entryName.endsWith(".SF");
+    }
+
+    @Override
+    public String toString() {
+        return "Policy(legacy)";
+    }
+
+    /**
+     * A no-op transformation that copies the original file verbatim. Can be used if the original cannot be instrumented under policy.
+     */
+    private static class SkipJarTransform implements JarTransform {
+
+        private static final Logger LOGGER = LoggerFactory.getLogger(InstrumentingClasspathFileTransformer.class);
```

Comment:
:nail_care: having a dependency on `InstrumentingClasspathFileTransformer` just for the logger seems excessive. Can we use something else?

Reply:
Good catch, we can just use SkipJarTransform